calico/felix/felix.py

```diff
@@ -643,44 +610,33 @@ def run(self):
                 #* messages.                                                 *#
                 #*************************************************************#
                 if sock.type == Socket.TYPE_EP_REP:
+                    #*********************************************************#
+                    #* We lost the connection on which we would receive      *#
+                    #* ENDPOINTCREATED messages. We may be out of step, so   *#
+                    #* need a total endpoint update.                         *#
+                    #*********************************************************#
                     endpoint_resync_needed = True
-                elif sock.type == Socket.TYPE_ACL_SUB:
+                elif (sock.type == Socket.TYPE_ACL_SUB or
+                      sock.type == Socket.TYPE_ACL_REQ):
+                    #*********************************************************#
+                    #* We lost the connection on which we would receive      *#
+                    #* ACLUPDATE messages, or might have lost some queued    *#
+                    #* GETACLSTATE messages. We may be out of step, so we    *#
+                    #* need a total ACL resync.                              *#
+                    #*********************************************************#
                     acl_resync_needed = True
-
-                # Flush the message queue.
-                if sock.type == Socket.TYPE_EP_REQ:
-                    self.endpoint_queue.clear()
-                elif sock.type == Socket.TYPE_ACL_REQ:
-                    self.acl_queue.clear()
+                elif sock.type == Socket.TYPE_EP_REQ:
+                    if self.resync_id is not None:
+                        #*****************************************************#
+                        #* The resync we were doing has officially           *#
+                        #* failed. We'll retry when the connection comes     *#
+                        #* back.                                             *#
+                        #*****************************************************#
+                        self.complete_endpoint_resync(False)
+                        endpoint_resync_needed = True
```

Comment:
We lost a message in the exchange to do with the resync; we need to tidy up that resync and try again. I tweaked the comment to this; let me know what you think.

                    #*********************************************************#
                    #* A resync was in progress, but we may have lost the    *#
                    #* RESYNCSTATE request or response or (potentially) an   *#
                    #* ENDPOINTCREATED message due to a lost                 *#
                    #* connection. That means we have to give up on this     *#
                    #* resync, tidy up and retry.                            *#
                    #*********************************************************#